Formula/totp-cli.rb

```diff
@@ -0,0 +1,19 @@
+class TotpCli < Formula
+  desc "Authy/Google Authenticator like TOTP CLI tool written in Go"
+  homepage "https://yitsushi.github.io/totp-cli/"
+  url "https://github.com/yitsushi/totp-cli/archive/v1.2.4.tar.gz"
+  sha256 "0451172dbd85ecddbe5faf14fecd7cefab3e8678ff6b8d6f5cb6cfa8d7defa3b"
+  license "MIT"
+  head "https://github.com/yitsushi/totp-cli.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    system "go", "build", "-o", bin/"totp-cli"
+  end
+
+  test do
+    assert_match("totp-cli v", shell_output("#{bin}/totp-cli version"))
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
test updated.